#! /bin/sh /usr/share/dpatch/dpatch-run
# Description: fix web interface memory disclosure
# From: Drew Yao

@DPATCH@
diff -urNad cups-1.4.3~/cgi-bin/var.c cups-1.4.3/cgi-bin/var.c
--- cups-1.4.3~/cgi-bin/var.c	2010-06-16 08:14:47.000000000 -0400
+++ cups-1.4.3/cgi-bin/var.c	2010-06-16 08:14:52.000000000 -0400
@@ -1111,6 +1111,9 @@
 	    * Read the hex code...
 	    */
 
+            if (!isxdigit(data[1] & 255) || !isxdigit(data[2] & 255))
+	      return (0);
+
             if (s < (value + sizeof(value) - 1))
 	    {
               data ++;
